File: src/lib/buster-api.ts
import * as request from 'superagent';

const BUSTER_API_URL = process.env.BUSTER_API_URL;

export enum BusterStatus {
  PENDING = 'PENDING',
  CANCELED = 'CANCELED',
  COMPLETED = 'COMPLETED',
  CREATED = 'CREATED',
}

export type BusterResponseType = {
  id: string,
  referenceId: string,
  status: BusterStatus,
  created: string,
}

export function createApiKey(webhookUrl: string) {
  return request
    .post(`${BUSTER_API_URL}/v1/api_key`)
    .send({ webhookUrl })
    .then(res => res.body);
}

export function createTransaction(referenceId: string, apiKey: string): Promise<BusterResponseType> {
  return request
    .post(`${BUSTER_API_URL}/v1/transaction`)
    .set({ ['X-API-KEY']: apiKey })
    .send({ referenceId })
    .then(res => res.body);
}

export function getTransaction(externalId: string, apiKey: string): Promise<BusterResponseType> {
  return request
    .get(`${BUSTER_API_URL}/v1/transaction/${externalId}`)
    .set({ ['X-API-KEY']: apiKey })
    .then(res => res.body);
}

export function getTransactionByReferenceId(referenceId: string, apiKey: string): Promise<BusterResponseType> {
  return request
    .get(`${BUSTER_API_URL}/v1/transaction`)
    .set({ ['X-API-KEY']: apiKey })
    .query({ referenceId })
    .then(res => res.body);
}

File: python/buster_api.py
import requests
import os
import sys

BUSTER_API_URL = os.getenv('BUSTER_API_URL', None)


def create_api_key(webhook_url):
    result = requests.post('{}/v1/api_key'.format(BUSTER_API_URL),
                           json={"webhookUrl": webhook_url})

    if result.status_code != 200:
        raise Exception(result.text)

    return result.json()


def create_transaction(reference_id, api_key):
    result = requests.post('{}/v1/transaction'.format(BUSTER_API_URL),
                           json={"referenceId": reference_id},
                           headers={'X-API-KEY': api_key})

    if result.status_code != 200:
        raise Exception(result.text)

    return result.json()


def get_transaction(external_id, api_key):
    result = requests.get('{}/v1/transaction/{}'.format(BUSTER_API_URL, external_id),
                          headers={'X-API-KEY': api_key})

    if result.status_code != 200:
        raise Exception(result.text)

    return result.json()


def get_transaction_by_reference_id(reference_id, api_key):
    result = requests.get('{}/v1/transaction?referenceId={}'.format(BUSTER_API_URL, reference_id),
                          headers={'X-API-KEY': api_key})

    if result.status_code != 200:
        raise Exception(result.text)

    return result.json()
